### ISSUE TITLE
Trigger `MemoizedInstanceVariableName` cop for `return @foo if defined? @foo`

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop-hq/rubocop#5717]
Thanks for #5445!

`@foo ||= value` is good, but not always: https://karolgalanciak.com/blog/2017/05/28/ruby-memoization-%7C%7C-equals-vs-defined-syntax/

So, I'm suggesting to add pattern `return @foo if defined? @foo` (and `defined?(@foo)`) for this cop.

/cc @satyap 

--------

## Expected behavior

`MemoizedInstanceVariableName` triggers for:

```ruby
def foo
  return @bar if defined? @bar
  @bar = false
end
```

## Actual behavior

`MemoizedInstanceVariableName` doesn't trigger for `defined?` with mismatched variable name.

## Steps to reproduce the problem

Code is above.

## RuboCop version

```
$ rubocop -V
0.54.0 (using Parser 2.5.0.4, running on ruby 2.5.0 x86_64-linux)
```

**Comment by oniofchaos** [comment]
I'm confused by your expected/actual behavior bits. It sounds like you're trying to say that your `foo` method should trigger a cop, but based on my reading of the `MemoizedInstanceVariableName` your `foo` method is using an instance variable `@foo` and therefore should pass. Therefore, the Actual behavior should also be the expected behavior.

**Comment by AlexWayfer** [comment]
> I'm confused by your expected/actual behavior bits. It sounds like you're trying to say that your foo method should trigger a cop, but based on my reading of the MemoizedInstanceVariableName your foo method is using an instance variable @foo and therefore should pass. Therefore, the Actual behavior should also be the expected behavior.

Thank you, good catch. You're right, I've fixed the description.

**Comment by AlexWayfer** [comment]
This is still actual.
